python-package/lightgbm/basic.py

```diff
@@ -122,7 +120,17 @@ def _load_lib() -> Optional[ctypes.CDLL]:
     return lib
 
 
-_LIB = _load_lib()
+# we don't need lib_lightgbm while building docs
+_LIB: ctypes.CDLL
```

Comment:
this declaration avoids the following from `mypy`:

```text
python-package/lightgbm/basic.py:132: error: Incompatible types in assignment (expression has type "CDLL", variable has type "_MockCDLL")
```

That warning is raised because without such declarations, `mypy` decides the type of an object based on the first assignment to it. This warning could also be avoided by switching the order of the `if` statement.